oeps/oep-0007.rst

```diff
@@ -0,0 +1,526 @@
+code as much as possible, while restricting most compatibility boilerplate to
+the file's imports.  To get an unambiguous text object, you do add ``from
+builtins import str``.  Most stdlib imports can just be imported using their
+Python 3 name, and future handles adding compatibility shims in the right place
+for Python 2.  Dictionaries still need to use a ``future.utils.viewitems(d)``
+shim, as there's no way to override dict literals.  The goal is to make the
+code look as much as possible like native Python 3 code, and to make the
+compatibility layer easy to remove when dropping Python 2 support.  The
+downside is that it is not always obvious when future is being used, which
+could lead to hard-to-debug issues when maintaining a cross-compatible future
+codebase.
+
+Our general recommendation is to use future if a repository can reasonably be
+converted and drop Python 2 support within a short time frame (roughly two
+months).  If the project cannot be converted quickly, or will need to maintain
+support for Python 2 for a while after conversion, six is a better option, as
+the explictness of the compatibility code will make it easier to find
+compatibility issues, and to avoid introducing new issues.
+
+Third-party dependencies
+------------------------
+
+Before converting a codebase to Python 3, we need to make sure the code we
+depend on will also support Python 3.  We can make a rough check for
+third-party library support using `Can I Use Python 3`_, either as a website or
+library.  This will sometimes show inaccurate
+results, as it depends upon self-reporting of library compatibility (via the
+library's ``setup.py`` classifiers), but will help guide our investigations and
+scope out the amount of work required.  Results can be tracked in the
+`Compatibility Audit wiki page`_.
+
+.. _Compatibility Audit wiki page: https://openedx.atlassian.net/wiki/display/ENG/Compatibility+Audit
+
+If a required library does not support Python 3, we have a few options:
+
+1. We can contribute a patch to support Python 3 to the library.
+2. We can request Python 3 support, and wait for the maintainers to implement it.
+3. We can find an alternative library that does support Python 3.
+
+Which path is best may depend on the enthusiasm of the maintainers for
+supporting Python 3, the amount of resources we want to commit to the project,
+and the availability and quality of alternatives.
+
+__future__ imports
+------------------
+
+All files should have the main ``__future__`` imports at the top to regularize
+some behaviors that differ by default between Python 2 and 3.
+
+* ``from __future__ import absolute_import`` prevents the use of implicit
+  relative imports
+* ``from __future__ import print_function`` makes print a function instead of a
+  statement.
+* ``from __future__ import division`` will make single-slash division
+  (``a / b``) always perform floating point division, and double-slash division
+  (``a // b``).
```

Comment:
Incomplete sentence. Should read `and double slash division ("a // b") perform integer division`